zipkin-collector/rabbitmq/src/test/java/zipkin2/collector/rabbitmq/ITRabbitMQCollector.java

```diff
@@ -76,16 +116,23 @@ void messageWithMultipleSpans(SpanBytesEncoder encoder) throws Exception {
   }
 
   /** Ensures malformed spans don't hang the collector */
-  @Test public void skipsMalformedData() throws Exception {
+  @Test void skipsMalformedData() throws Exception {
+    RabbitMQCollector.Builder builder = builder("decoder_exception");
+
     byte[] malformed1 = "[\"='".getBytes(UTF_8); // screwed up json
     byte[] malformed2 = "malformed".getBytes(UTF_8);
-    publish(THRIFT.encodeList(spans));
-    publish(new byte[0]);
-    publish(malformed1);
-    publish(malformed2);
-    publish(THRIFT.encodeList(spans));
-
-    Thread.sleep(200L);
+    produceSpans(THRIFT.encodeList(spans), builder.queue);
+    produceSpans(new byte[0], builder.queue);
+    produceSpans(malformed1, builder.queue);
+    produceSpans(malformed2, builder.queue);
+    produceSpans(THRIFT.encodeList(spans), builder.queue);
+
+    try (RabbitMQCollector collector = builder.build()) {
+      collector.start();
+      assertThat(receivedSpans.take()).containsExactlyElementsOf(spans);
+      // the only way we could read this, is if the malformed spans were skipped.
+      assertThat(receivedSpans.take()).containsExactlyElementsOf(spans);
```

Comment:
It's slightly nerve-racking to use the same `spans` for both. But that's not special to this PR anyways

Reply:
this nerve racking repeats in a bunch of places, but I'll skip redoing that for now. agreed anyway